libretroshare/src/retroshare/rsmsgs.h

```diff
@@ -93,9 +95,31 @@ const ChatLobbyFlags RS_CHAT_LOBBY_FLAGS_PGP_SIGNED    ( 0x00000010 ) ; // requi
 typedef uint64_t	ChatLobbyId ;
 typedef uint64_t	ChatLobbyMsgId ;
 typedef std::string ChatLobbyNickName ;
-typedef std::string RsMailMessageId;			// should be uint32_t !!
-typedef uint64_t    MessageId ;
+typedef std::string RsMailMessageId; // TODO: rebase on t_RsGenericIdType
 
+/**
+ * Used to return a tracker id so the API user can keep track of sent mail
+ * status
+ */
+struct RsMailTrackId : RsSerializable
+{
+	RsMailTrackId(RsMailMessageId mailId, RsGxsId recipientId):
+	    mMailId(mailId), mRecipientId(recipientId) {}
+
+	RsMailMessageId mMailId;
+	RsGxsId mRecipientId;
```

Comment:
seems that only mMailId is needed, since the rest of information (msg content, flags, and recipitent) should be available by using getMessage. It's inconsistent to add the recipient *inside* the ID structure.

Reply:
The problem is that when you use getMail to check the status of a mail you sent, you get the mail with all the recipients, you have no way to check to which specific recipient that status apply. Instead passing back both mail id and the specific recipient for each single mail sent the caller can keep track of the status for each single recipient, becaus she knows each mail id to which specific sender has been sent.

Reply:
Then in the sendMail function, you may return a std::map<MailId, Recipient ID>. It feels a bit inconsistent to mix the two in the Mail ID structure, especially if you only need this in one particular place in the code.

Reply:
be aware that It is not Mail id structure ;)